submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/integration/PersistenceContextImpl.kt

```diff
@@ -59,6 +62,10 @@ open class PersistenceContextImpl(
             subRepository.expireActiveVersions(submission.accNo)
             submission.version = nextVersion
             subRepository.save(subMapper.toSubmissionDb(submission))
+            // TODO: Refactor draft deletion by making DraftService listen to a successful submission event
+            userDataRepository.findByUserIdAndKeyIgnoreCaseContaining(submission.user.id, submission.accNo).ifNotEmpty {
```

Comment:
I don't think we should place this directly here. There's already a method for that in `SubmissionDraftService`. The proper way of doing this should be placing it as a new step in the submission process in `SubmissionSubmitter`

Reply:
We want to carry out this particular operation in the same transaction to avoid multiple submissions by accident (due to lag or slow response of backend/proxy). As pointed out by @Juan-EBI , we can't guarantee consistency with distributed transactions so in this particular case, I think we should fix this here temporarily until we can discuss a better approach in detail. This will be a minor addition to the technical debt but a major issue would be closed.

Reply:
but having this here is a hidden effect. Basically a function that is call `getSequenceNextValue` also delete drafts. What I suggest we do here is to solve this,  is moving `@Transactional` to `SubmissionSubmitter` and follow @jhoanmanuelms advice.

Reply:
I'm sold at the business logic point. We can make `SubmissionSubmitter.submit()` transactional but `SubmissionDraftService` is in the `submission-webapp` module. How about moving the logic to `SubmissionService.submit()` instead and making that `@Transactional?`

Reply:
Scratch that. It creates a circular dependency. Let me try to refactor the logic in `PersistenceContext` and see if that makes more sense.